lib/AST/GenericSignature.cpp

```diff
@@ -396,6 +396,12 @@ GenericSignature::getSubstitutionMap(SubstitutionList subs) const {
         assert(reqts[i].getSecondType()->getAnyNominal() ==
                conformances[i].getRequirement());
         result.addConformance(canTy, conformances[i]);
+
+        auto canReplacement = replacement->getCanonicalType();
+        if (!canReplacement->isTypeParameter() &&
+            !canReplacement->is<ArchetypeType>()) {
+          result.addConformance(canReplacement, conformances[i]);
```

Comment:
It surprises me that we need this.

Reply:
We don't. Vestigates of a hack that I seemingly didn't fully remove.